compiler/ccgexprs.nim

```diff
@@ -1204,7 +1204,11 @@ proc genObjConstr(p: BProc, e: PNode, d: var TLoc) =
     if handleConstExpr(p, e, d): return
   var tmp: TLoc
   var t = e.typ.skipTypes(abstractInst)
-  getTemp(p, t, tmp)
+  let noTemp = d.k in {locTemp,locLocalVar,locGlobalVar,locParam,locField}
```

Comment:
I doubt this is correct, what if I do ``x = Obj(field: x)``? (That's valid for ``ref object`` at least.)